lib/utmp.c

```diff
@@ -275,7 +294,7 @@ prepare_utmp(const char *name, const char *line, const char *host,
 	utent->ut_type = USER_PROCESS;
 	utent->ut_pid = main_pid;
 	STRNCPY(utent->ut_line, line);
-	if (NULL != ut) {
+	if ((NULL != ut) && (0 != utent->ut_id[0])) {
```

Comment:
Please use `'\0'` for the null character.

Reply:
I can only repeat myself.
Without any written guidelines every contributor will have to re-write each PR several times, without a chance to guess expected format at the first try.